library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java

```diff
@@ -226,19 +224,12 @@ private void configAttachmentUI() {
             }
             Utils.setButtonDelayEnable(v);
             messageInputController.onClickCloseBackGroundView();
-
-            Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-            ContentValues values = new ContentValues();
-            values.put(MediaStore.Images.Media.TITLE, "New Picture");
-            values.put(MediaStore.Images.Media.DESCRIPTION, "From your Camera");
-            imageUri = getContext().getContentResolver().insert(
-                    MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values);
-            takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
-
-            Intent takeVideoIntent = new Intent(MediaStore.ACTION_VIDEO_CAPTURE);
+            Intent takePictureIntent = CaptureController.getTakePictureIntent(getContext());
+            Intent takeVideoIntent = CaptureController.getTakeVideoIntent(getContext());
             Intent chooserIntent = Intent.createChooser(takePictureIntent, "Capture Image or Video");
             chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[]{takeVideoIntent});
-
+            StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
```

Comment:
Hm, whats the purpose of `StrictMode.VmPolicy.Builder` here?

Reply:
We have changed the logic of taking image and recording video, that is, we should save captured files to special folder(for example: `storage/app name/image or video`) so indicates file to the takePictureIntent.
In this case we need this lines.

https://stackoverflow.com/questions/48117511/exposed-beyond-app-through-clipdata-item-geturi